servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/apple/traffic/resilience/http/TrafficResilienceHttpServiceFilter.java

```diff
@@ -0,0 +1,688 @@
+            return this;
+        }
+
+        /**
+         * {@link Ticket Ticket} terminal callback override upon cancellation of the operation.
+         * By default, the terminal callback is {@link Ticket#ignored()}.
+         * <p>
+         * You may need to adjust this callback depending on the ordering this filter was applied.
+         * For example if the filter is applied after the
+         * {@link io.servicetalk.http.utils.TimeoutHttpRequesterFilter timeout-filter} then you may want to also
+         * {@link Ticket#dropped() drop the ticket} to let the algorithm apply throttling accounting for this timeout.
+         * @param onCancellation Callback to override default {@link Ticket ticket} terminal event when an operation
+         * is cancelled.
+         * @return {@code this}.
+         */
+        public Builder onCancelTicketTerminal(final Consumer<Ticket> onCancellation) {
+            this.onCancellationTicketTerminal = requireNonNull(onCancellation, "onCancellation");
+            return this;
+        }
+
+        /**
+         * Defines the {@link RejectionPolicy} which in turn defines the behavior of the service when a
+         * rejection occurs due to {@link CapacityLimiter capacity} or {@link CircuitBreaker breaker}.
+         *
+         * @param policy The policy to put into effect when a rejection occurs.
+         * @return {@code this}.
+         */
+        public Builder onRejectionPolicy(final RejectionPolicy policy) {
+            this.onRejectionPolicy = requireNonNull(policy, "policy");
+            return this;
+        }
+
+        /**
+         * Provide an observer to track interactions of the filter and requests.
+         * @param observer an observer to track interactions of the filter and requests.
+         * @return {@code this}.
+         */
+        public Builder observer(final TrafficResiliencyObserver observer) {
+            requireNonNull(observer, "observer");
+            this.observer = new SafeTrafficResiliencyObserver(observer);
+            return this;
+        }
+
+        /**
+         * Invoke to build an instance of {@link TrafficResilienceHttpServiceFilter} filter to be used inside the
+         * {@link HttpServerBuilder}.
+         * @return An instance of {@link TrafficResilienceHttpServiceFilter} with the characteristics
+         * of this builder input.
+         */
+        public TrafficResilienceHttpServiceFilter build() {
+            return new TrafficResilienceHttpServiceFilter(capacityPartitionsSupplier, rejectNotMatched,
+                    classifier, onCompletionTicketTerminal, onCancellationTicketTerminal,
+                    onErrorTicketTerminal, circuitBreakerPartitionsSupplier, onRejectionPolicy, observer);
+        }
+    }
+
+    /**
+     * Policy to rule the behavior of service rejections due to capacity or open circuit.
+     */
+    public static final class RejectionPolicy {
```

Comment:
Similar to `PeerCapacityRejectionPolicy`, consider making it a top level class. In its javadoc, consider referencing where it can be used.